libs/lua-hash/src/hash.c

```diff
@@ -0,0 +1,47 @@
+#include <libubox/md5.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MD5_DIGEST_LEN 16
+#define MD5_STRLEN 2*MD5_DIGEST_LEN+1
+
+static int md5(lua_State *L) {
+	char *input = (char *) lua_tostring(L, 1);
+	if (input == NULL) {
+		lua_pushstring(L, "first argument: should be the input string");
+		lua_error(L);
+		return 1;
```

Comment:
The return is superfluous, as lua_error will longjmp. It should probably be replaced with `__builtin_unreachable()`, as Lua doesn't define the function with attribute((noreturn)).